projects/rpx-xui-translation/src/lib/helpers/replace-placeholders/replace-placeholders.helper.ts

```diff
@@ -16,29 +16,54 @@ export const replacePlaceholders = (input: string, replacements: Replacements) =
 export const splitPhraseIntoComponents = (phrase: string, replacements: Record<string, string>): string[] => {
   const result: string[] = [];
   const placeholders = Object.keys(replacements).map((key) => `%${key}%`);
+  phrase = phrase.trim();
 
   // Regular expression to match all placeholders in the phrase
   const placeholderRegex = new RegExp(placeholders.join('|'), 'g');
 
-  // Split the phrase into parts based on the placeholders
-  const parts = phrase.split(placeholderRegex);
+  let lastIndex = 0;
 
-  // Match placeholders in the phrase for adding their replacement values
-  const matches = phrase.match(placeholderRegex) || [];
+  // Helper function to split text into spaces and content
+  const splitTextIntoComponents = (text: string): string[] => {
+    const components: string[] = [];
+    const trimmed = text.trim();
 
-  // Iterate through parts and matches to build the result array
-  parts.forEach((part, index) => {
-    const trimmedPart = part.trim();
+    if (text.startsWith(' ')) {
```

Comment:
You need to handle multiple spaces - possibly OK for now, but just put a loop around it, and also for trailing space